sycl/include/sycl/detail/stl_type_traits.hpp

```diff
@@ -17,34 +17,9 @@ namespace sycl {
 __SYCL_INLINE_VER_NAMESPACE(_V1) {
 namespace detail {
 
-// Type traits identical to those in std in newer versions. Can be removed when
-// SYCL requires a newer version of the C++ standard.
-// C++14
-template <bool B, class T, class F>
-using conditional_t = typename std::conditional<B, T, F>::type;
-
-template <typename T>
-using remove_const_t = typename std::remove_const<T>::type;
-
-template <typename T> using remove_cv_t = typename std::remove_cv<T>::type;
-
-template <typename T>
-using remove_reference_t = typename std::remove_reference<T>::type;
-
-template <typename T> using add_pointer_t = typename std::add_pointer<T>::type;
-
-// C++17
-template <bool V> using bool_constant = std::integral_constant<bool, V>;
-
-template <class...> using void_t = void;
-
-// Custom type traits
-template <typename T>
-using allocator_value_type_t = typename std::allocator_traits<T>::value_type;
-
-template <typename T>
-using allocator_pointer_t = typename std::allocator_traits<T>::pointer;
-
+// Custom type traits.
+// FIXME: Those doesn't seem to be a part of any published/future C++ standard
+// so should probably be moved to a different place.
```

Comment:
We probably can drop `stl_` prefix from the file name to avoid confusion. What do you think?

Reply:
We have `detail/type_traits.hpp` already that has lots of custom stuff (and `remove_pointer` that seems to behave differently than `std::` version). I think that the best would be to move this code to `detail/type_traits.hpp`.

As for this file, I'm planning to add C++20's `remove_cvref_t` here.

Reply:
Following up on the previous comment - `sycl::detail::remove_pointer` difference comes from `sycl::multi_ptr` support, I think.